core/math/geometry_3d.h

```diff
@@ -41,6 +41,9 @@ class Geometry3D {
 	static void get_closest_points_between_segments(const Vector3 &p_p0, const Vector3 &p_p1, const Vector3 &p_q0, const Vector3 &p_q1, Vector3 &r_ps, Vector3 &r_qt);
 	static real_t get_closest_distance_between_segments(const Vector3 &p_p0, const Vector3 &p_p1, const Vector3 &p_q0, const Vector3 &p_q1);
 
+	static void get_closest_points_between_line_and_circle(const Vector3 &p_line_origin, const Vector3 &p_line_direction, const Vector3 &p_circle_center, const Vector3 &p_circle_normal, const real_t p_circle_radius, Vector<Vector3> &r_line_closest, Vector<Vector3> &r_circle_closest, size_t &r_num_closest_pairs, bool &r_equidistant);
```

Comment:
I would change this function to not return Vector, since memory allocation is generally very inefficient in critical code like this. Probably better to have something like `Vector3 * r_circle_closest`
and if there are many results, then a `int &r_circle_closest_count`